tests/test_persistence.py

```diff
@@ -52,3 +52,44 @@ def test_libsecret_persistence(temp_location):
         {"my_attr_1": "foo", "my_attr_2": "bar"},
         ))
 
+@pytest.mark.skipif(
+    is_running_on_travis_ci, reason="Requires manual testing")
+def test_windows_empty_file_exists_before_first_use():
+    test_folder = tempfile.mkdtemp(prefix="msal_extension_test_windows_token_cache_roundtrip")
+    cache_file = os.path.join(test_folder, 'msal.cache')
+    open(cache_file, 'w+')
+    try:
+        persistence = FilePersistenceWithDataProtection(cache_file)
+        assert persistence.load() is None  # Empty file should just be a NO-OP
+    finally:
+        shutil.rmtree(test_folder, ignore_errors=True)
+
+@pytest.mark.skipif(
+    is_running_on_travis_ci, reason="Requires manual testing")
+def test_windows_file_contains_data_before_first_use():
+    test_folder = tempfile.mkdtemp(prefix="msal_extension_test_windows_token_cache_roundtrip")
+    cache_file = os.path.join(test_folder, 'msal.cache')
+    fh = open(cache_file, 'w+')
+    cache_content = '{"AccessToken": {}, "Account": {}, "IdToken": {}, "RefreshToken": {}, "AppMetadata": {}}'
+    fh.write(cache_content)
+    fh.close()
+    try:
+        persistence = FilePersistenceWithDataProtection(cache_file)
+        with pytest.raises(OSError) as err:
+            persistence.load()
+        assert err.value.winerror == 13  # WinError 13 - The data is invalid
+    finally:
+        shutil.rmtree(test_folder, ignore_errors=True)
+
+@pytest.mark.skipif(
+    is_running_on_travis_ci, reason="Requires manual testing")
```

Comment:
Have you run these test cases locally? Wouldn't it always fail one way or another because they ALL try to run regardless whether you are on Windows or Mac?

Reply:
I just have to run them separately not as a file, so basically just run the windows one on a windows and macOS ones on Mac

Reply:
But if you try to run ALL of them at once it some cases will fail depending on the platform

Reply:
Added `skipif` condition to make sure the tests are skipped based on platform in [this](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/pull/65/commits/fab3c72576419c8a839a352f99e18b489109f297) commit